scipy/stats/_mstats_basic.py

```diff
@@ -1148,15 +1149,114 @@ def siegelslopes(y, x=None, method="hierarchical"):
     return stats_siegelslopes(y, x, method=method)
 
 
+SenSeasonalSlopesResult = _make_tuple_bunch('SenSeasonalSlopesResult',
+                                            ['intra_slope', 'inter_slope'])
+
+
 def sen_seasonal_slopes(x):
+    r"""
+    Computes seasonal Theil-Sen and Kendall slope estimators.
+
+    The seasonal generalization of Sen's slope computes the slopes between all
+    pairs of values within a "season" (column) of a 2D array. It returns an
+    array containing the median of these "within-season" slopes for each
+    season (the Theil-Sen slope estimator of each season), and it returns the
+    median of the within-season slopes across all seasons (the seasonal Kendall
+    slope estimator).
+
+    Parameters
+    ----------
+    x : 2D array_like
+        Each column of `x` contains measurements of the dependent variable
+        within a season. The independent variable of each season is assumed
+        to be ``np.arange(x.shape[0])``.
+
+    Returns
+    -------
+    result : ``SenSeasonalSlopesResult`` instance
+        The return value is an object with the following attributes:
+
+        intra_slope : ndarray
+            For each season, the Theil-Sen slope estimator: the median of
+            within-season slopes.
+        inter_slope : float
+            The seasonal Kendall slope estimateor: the median of within-season
+            slopes *across all* seasons.
+
+    See Also
+    --------
+    theilslopes : the analogous function for non-seasonal data
+
+    Notes
+    -----
+    The slopes :math:`d_{ijk}` within season :math:`i` are:
+
+    .. math::
+
+        d_{ijk} = \frac{x_{ij} - x_{ik}}
+                            {j - k}
+
+    for pairs of distinct integer indices :math:`j, k` of :math:`x`.
+
+    Element :math:`i` of the returned `intra_slope` array is the median of the
+    :math:`d_{ijk}` over all :math:`j < k`; this is the Theil-Sen slope
+    estimator of season :math:`i`. The returned `inter_slope` value, better
+    known as the seasonal Kendall slope estimator, is the median of the
+    :math:`d_{ijk}` over all :math:`i, j, k`.
+
+    References
+    ----------
+    .. [1] Hirsch, Robert M., James R. Slack, and Richard A. Smith.
+           "Techniques of trend analysis for monthly water quality data."
+           *Water resources research* 18.1 (1982): 107-121.
+
+    Examples
+    --------
+    Suppose we have 100 observations of a dependent variable for each of 10
```

Comment:
it might be easier to take 4 seasons (quarters) or 5 (Mon-Fri) to make the example for intuitive. That way, we can also print the results below (next comment)

Reply:
It's a great dea, and I wanted to do something like it, but I would have wanted to use real data with something interesting to say about the slopes and seasons. Also, by using real data, we wouldn't have to worry about the mysterious seed of the RNG, which always makes it difficult (for me, at least) to keep the output correct, and then the results would be exactly reproducible by the reader.
However, I ran out of the time I allotted to this : /
If you'd be willing to do an enhancement like that, I'd be happy to review!

Reply:
a real example would be even better (I don't have time at the moment). I just thought the example would already be easier to understand even with random data if one prints the results and adds a comment that the slopes are expected to be close to zero for random data. if you do that, it is probably better to reduce the number of columns to avoid printing too many numbers. but you can also keep it as it is

Reply:
Thanks. I don't think I should spend much more time on this either right now. I just set out to close that old issue. If you think this closes the issue as-is, let's merge it. We can come back when some of the other high priority issues are resolved.

Reply:
I went ahead and added this : )

Reply:
I like the updated version. I will resolve this conversation